src/main/java/graphql/nadel/engine/AsyncIsFieldForbidden.java

```diff
@@ -32,9 +32,32 @@ public AsyncIsFieldForbidden(ServiceExecutionHooks serviceExecutionHooks, NadelC
     }
 
     public CompletableFuture<Map<String, GraphQLError>> getForbiddenFields(Node<?> root) {
-        return visitChildren(root).thenApply(unused -> {
-            return fieldIdsToErrors;
-        });
+        return CompletableFuture.allOf(
```

Comment:
we should not visit the children if the top level field is remove. The current code does.